python/openchange/provision.py

```diff
@@ -775,12 +776,38 @@ def newgroup(names, lp, creds, groupname, mail=None,):
                  be set to <samAccountName>@<dnsdomain>
     """
     db = get_local_samdb(names, lp, creds)
-    group_dn = get_group_dn(db, names.domaindn, names.firstorg, groupname)
-
+    group_dn = get_group_dn(db, names.domaindn, groupname)
     if group_dn:
-        smtp_user, mail_domain = _smtp_user_and_domain_by_dn(group_dn, groupname, mail, names)
-        (recipient_type_details, recipient_display_type) = _group_recipient_type_details(db, group_dn)
-        extended_group = """
+        _enable_group(db, names, group_dn, groupname, mail)
+        print "[+] Group %s extended and enabled" % groupname
+    else:
+        print "[!] Group '%s' not found" % groupname
+
+
+def enable_all_groups_in_organization(names, lp, creds, org_name):
+    """enable all groups in the given organization
+
+    Groups without mail atrtibute are ignored
+    This methods only works in schemas where the groups reside under "CN=Groups,CN=%s,CN=Users,%s" % (org_name, domain_dn)
+    """
+    basedn = "CN=Groups,CN=%s,CN=Users,%s" % (org_name, names.domaindn)
+    ldap_filter = "(&(objectClass=group)(sAMAccountName=*@%s)(mail=*))" % (org_name)
```

Comment:
`(sAMAccountName=*@%s)` should not be needed. You are already filtering by organization with the base dn so I think is better not to assume the sAMAccountName is following that format (we are not enforcing any format on creation)